src/main/java/david/task/Task.java

```diff
@@ -0,0 +1,61 @@
+package david.task;
+
+/**
+ * Abstract task class
+ */
+public abstract class Task {
```

Comment:
I like how you made the Task class an abstract class since it wouldn't be instantiated.